hrm-domain/hrm-core/case/caseService.ts

```diff
@@ -94,6 +93,9 @@ export type CaseService = CaseRecordCommon & {
   categories: Record<string, string[]>;
   precalculatedPermissions?: PrecalculatedPermissions;
   connectedContacts?: Contact[];
+  sections: {
+    [k in (typeof WELL_KNOWN_CASE_SECTION_NAMES)[keyof typeof WELL_KNOWN_CASE_SECTION_NAMES]['sectionTypeName']]?: CaseSection[];
+  };
```

Comment:
This looks like a very complex inline type definition.
Would it be worth to extract this section `type` and give it a meaningful name?